libwallet/internal/loader/dcr/loader.go

```diff
@@ -50,6 +50,10 @@ type dcrLoader struct {
 	mixSplitLimit           int
 
 	mu sync.RWMutex
+
+	// dbMutex should be held when db transactions would circle back around
+	// and hold the mu lock to prevent a freeze.
+	dbMutex *sync.Mutex
```

Comment:
can we use mu for all? I mean pass mutex from wallet.go to dcr loader

Reply:
No. As the comment says below `This allows callers to call *dcrLoader methods inside of db transactions.`